aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -85,12 +85,18 @@ Tensor& _clamp_min_out_cpu(Tensor& result, const Tensor& self, Scalar min) {
   return legacy::th::_th_clamp_min_out(result, self, min);
 }
 
+Tensor& fill_out(Tensor& self, const Scalar value) {
+  auto iter = TensorIterator::unary_op(self, self);
```

Comment:
Question: should this be a nullary op instead?

Reply:
I wasn't very confident and I'm pretty open to the idea that nullary might be preferable. We don't currently have any `NullaryOpsKernel` etc.

Unary seems to make sense semantically, in a file with operations like abs() etc.

The macros that provide TH implementations for some of the other unary ops also seems somewhat similar with the (self, self) for in-place operations, e.g.:
```
  Tensor& _##op##__cpu(Tensor& self) {                          \
    return at::op##_out(self, self);                            \
  }                                                             \
```

Another option is that it could be clearer as unary if the second arg is a Tensor, called by the Scalar implementation by using wrapped_scalar_tensor(scalar_value) as is done in many of the BinaryOps. I felt like if I converted to Tensor I'd just need to get back to a Scalar anyway though, so didn't take that path.

Reply:
You should use `TensorIterator::nullary_op` since this only writes self and doesn't read self. It's fine to keep the definition in this file -- there's a reasonable semantic argument that this is a "unary" operation even if TensorIterator doesn't care about `Scalar value`.

I agree that it's probably best not to use wrapped_scalar_tensor here.

This will require implementing a few new functions in Loops.h: `nullary_kernel_vec`, `nullary_kernel`, and their transitive dependencies. You should be able to grab them from @cpuhrsch's PR: https://github.com/pytorch/pytorch/pull/18876/files#diff-0d1178f1a4ce15aeb760d251974e6924

On the GPU side, there is already a `gpu_nullary_kernel`.

Reply:
I was thinking of it as total inputs to create the result (need a value to fill with), which is why I was thinking of it as unary, but, will also be easier to implement with nullary now that I can grab the loop from that PR. Thanks.

Reply:
looking more closely at the linked PR, I didn't realize someone was already working on fill_. I was looking at it because of this PR: https://github.com/pytorch/pytorch/pull/20336#issuecomment-493260729 and also just wanted to understand TensorIterator better.